runtime/lua/vim/ui.lua

```diff
@@ -55,6 +55,90 @@ function M.select(items, opts, on_choice)
   end
 end
 
+--- Prompts the user to pick one or multiple items from a collection of entries
+---
+---@param items table Arbitrary items
+---@param opts table Additional options
+---     - prompt (string|nil)
+---               Text of the prompt. Defaults to `Select from list or press return to confirm: `
+---     - format_item (function item -> text)
+---               Function to format an
+---               individual item from `items`. Defaults to `tostring`.
+---     - kind (string|nil)
+---               Arbitrary hint string indicating the item shape.
+---               Plugins reimplementing `vim.ui.select_many` may wish to
+---               use this to infer the structure or semantics of
+---               `items`, or the context in which select_many() was called.
+---@param on_choices function ((chosen_items|nil, indexes|nil) -> ())
+---               Called once the user made a choice.
+---               `indexes` is a list of 1-based indexes of `chosen_items` within `items`.
+---               `nil` if the user aborted the dialog without selecting any items.
+---
+---
+--- Example:
+--- <pre>
+--- vim.ui.select_many({ 'cheese', 'pepperoni', 'anchovies' }, {
+---     prompt = 'Select your toppings:',
+---     format_item = function(item)
+---         return "some " .. item
+---     end,
+--- }, function(choices)
+---     if choices then
+---         for _, item in ipairs(choices) do
+---             -- add topping to pizza or whatever
+---         end
+---     end
+--- end)
```

Comment:
Fun example, but can you think of something perhaps more directly related to something one might do in Neovim? For example the docs for `ui.select` show an example of choosing between tabs and spaces. Something in that same spirit would be better, I think.

Reply:
perhaps enable some options?
e.g. number, relative number and cursorline